### PR TITLE
Js Task 7

### PR DESCRIPTION
**Pull request by dudzon**
https://mogoslider.netlify.com/

### CODE REVIEW

File: css_tasks/task11/js/slider.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import {constants} from './constants.js';
+export function slider() {
+  elements.then(
+      ({
+        firstCarousel,
+        secondCarousel,
+        firstCarouselItems,
+        secondCarouselItems,
+      }) => {
+        let {slideIndex} = constants;
```
Comment (tkachukivan) [inline css_tasks/task11/js/slider.js:15]: you change this value, it is not constant, just variable with some default value

File: css_tasks/task11/js/slider.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        const handleLeftArrow = (e, carouselItems) => {
+          if (!e.target.classList.contains('decoration-carousel__arrow--left')) {
+            return;
+          }
+          handleSlides(-1, carouselItems, 'backward');
+        };
+
+        // Move slides
+        const handleSlides = (n, items, direction) => {
+          showSlides((slideIndex += n), items, direction);
```
Comment (tkachukivan) [inline css_tasks/task11/js/slider.js:32]: better to have this logic `(slideIndex += n)` in function

File: css_tasks/task11/js/slider.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          : items[slideIndex - 1].classList.add('backward');
+        };
+
+        // EVENTS
+
+        firstCarousel.addEventListener('click', (e) =>
+          handleRightArrow(e, firstCarouselItems)
+        );
+        firstCarousel.addEventListener('click', (e) =>
+          handleLeftArrow(e, firstCarouselItems)
```
Comment (tkachukivan) [inline css_tasks/task11/js/slider.js:61]: why not to do it with just one listener, for example with saving in data items direction attribute, same below

File: css_tasks/task11/js/slider.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+          if (n > items.length) {
+            slideIndex = 1;
+          }
+          if (n < 1) {
+            slideIndex = items.length;
+          }
+          for (i = 0; i < items.length; i++) {
+          // it's not superhandy, should be rewritten somehow :)
+            items[i].classList.remove('active', 'forward', 'backward');
```
Comment (tkachukivan) [inline css_tasks/task11/js/slider.js:45]: save active element and remove from this element

**Review by tkachukivan — COMMENTED** [review]
it is perfect place try to use some generic function/class to which you will just need to pass element(root/maybe also list of items) and it will do all work, you will just need to call it twice(create two instances) with different arguments

File: css_tasks/task11/js/slider.js (hunk 1 above)
Comment (tsvetkovv) [reply css_tasks/task11/js/slider.js:15]: Moreover, it's declaration of new variable with the same name. The initial `slideIndex` in the file is unchanged.

File: css_tasks/task11/js/slider.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          }
+          handleSlides(-1, carouselItems, 'backward');
+        };
+
+        // Move slides
+        const handleSlides = (n, items, direction) => {
+          showSlides((slideIndex += n), items, direction);
+        };
+        const showSlides = (n, items, direction) => {
+          let i;
```
Comment (tsvetkovv) [inline css_tasks/task11/js/slider.js:35]: think how to set slideIndex by the modulo operator :)

File: css_tasks/task11/js/slider.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+          if (n > items.length) {
+            slideIndex = 1;
+          }
+          if (n < 1) {
+            slideIndex = items.length;
+          }
+          for (i = 0; i < items.length; i++) {
+          // it's not superhandy, should be rewritten somehow :)
```
Comment (tsvetkovv) [inline css_tasks/task11/js/slider.js:44]: try to leave only forward of backward

File: css_tasks/task11/scss/_decoration.scss [hunk 7]
```diff
@@ -109,16 +107,67 @@
...
+  animation: fade-backward;
+  animation-duration: 1.5s;
+}
+.decoration__wrapper--carousel-first
+  .carousel-first
+  .decoration-carousel__text:after,
+.decoration__wrapper--carousel-second
+  .carousel-first
+  .decoration-carousel__text:after {
+  content: "John Doe";
```
Comment (tsvetkovv) [inline css_tasks/task11/scss/_decoration.scss:132]: Try to leave content only in HTML
https://developer.mozilla.org/en-US/docs/Web/CSS/attr#content_property

Comment (tsvetkovv) [reply css_tasks/task11/scss/_decoration.scss:132]: and then you don't need to duplicate this in CSS

File: css_tasks/task11/js/slider.js (hunk 5 above)
Comment (tsvetkovv) [reply css_tasks/task11/js/slider.js:35]: oh, % is not modulo. never mind

**Pull request merged**